src/web/app/pages-student/student-course-details-page/student-course-details-page.component.ts

```diff
@@ -52,67 +24,119 @@ interface StudentCourseDetails {
 
+  /**
+   * Fetches relevant data to be displayed on page.
+   */
   ngOnInit(): void {
     this.route.queryParams.subscribe((queryParams: any) => {
       this.user = queryParams.user;
-      this.loadStudentDetails(queryParams.courseid, this.user);
+      this.loadStudent(queryParams.courseid);
+      this.loadCourse(queryParams.courseid);
+      this.loadInstructors(queryParams.courseid);
     });
   }
 
   /**
-   * Fetch the data to be displayed on the page.
+   * Loads the course details.
    * @param courseid: id of the course queried
-   * @param user: only used in admin masquerade mode, value should be student id
    */
-  loadStudentDetails(courseid: string, user: string): void {
-    const paramMap: { [key: string]: string } = { courseid, user };
-    this.httpRequestService.get('/student/course', paramMap).subscribe((resp: StudentCourseDetails) => {
-      this.student = resp.student;
-      this.instructorDetails = resp.instructorDetails;
-      this.course = resp.course;
-      this.teammateProfiles = resp.teammateProfiles;
-
-      if (!this.student) {
-        this.statusMessageService.showErrorMessage('Error retrieving student details');
-      }
-
-      if (!this.course) {
-        this.statusMessageService.showErrorMessage('Error retrieving course details');
-      }
+  loadCourse(courseid: string): void {
+    this.courseService.getCourse(courseid).subscribe((course: Course) => {
+      this.course = course;
+    });
+  }
 
-      if (!resp.instructorDetails) {
-        this.statusMessageService.showErrorMessage('Error retrieving instructor details');
-      }
+  /**
+   * Loads the students of the course.
+   * @param courseid: id of the course queried
+   */
+  loadStudent(courseid: string): void {
+    const paramMap: { [key: string]: string } = {
+      courseid,
+    };
+
+    this.httpRequestService.get('/student', paramMap)
+        .subscribe((student: Student) => {
+          this.student = student;
+          this.loadTeammates(courseid, student.teamName);
+        }, (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorMessage(resp.error.message);
+        });
+  }
 
-      if (!this.teammateProfiles) {
-        this.statusMessageService.showWarningMessage('You do not have any teammates yet.');
-      }
-    }, (resp: ErrorMessageOutput) => {
-      this.statusMessageService.showErrorMessage(resp.error.message);
-    });
+  /**
+   * Loads the teammates of the current student.
+   * @param courseid: id of the course queried
+   * @param teamName: team of current student
+   */
+  loadTeammates(courseid: string, teamName: string): void {
+    this.studentService.getStudentsFromCourseAndTeam(courseid, teamName)
+        .subscribe((students: Students) => {
+          students.students.forEach((student: Student) => {
+            // filter away current user
+            if (this.student && student.email === this.student.email && student.name === this.student.name) {
```

Comment:
- Is this check necessary `this.student`?
- I think just checking email is enough.

Reply:
Yup yourr right. Fixed